src/features/quick-search/QuickSearchItem.js

```diff
@@ -16,6 +16,28 @@ import './QuickSearchItem.scss';
 
 const QUERY_SEPARATOR = '<mark>';
 
+type QuickSearchItemData = {
+    extension?: string,
+    iconType: string,
+    id: string,
+    name: string,
+    nameWithMarkedQuery: string,
+    parentName?: string,
+    sharedLink?: string,
+    type: ItemType,
+    updatedBy: string,
+    updatedDate: number,
+};
+
+type Props = {
+    className?: string,
+    closeDropdown?: Function,
+    intl: IntlShape,
+    itemData: QuickSearchItemData,
```

Comment:
This can be an Immutable record of type QuickSearchItemData. I couldn't quite get the flow typing to work with that. I tried importing the Record type from immutable.js and it didn't seem to work

Reply:
Does the addition of flow to this component cause any issues for the consuming applications? If so, this commit will need to include `BREAKING CHANGE: <your change>` in the commit body and you'll need to open PRs for each project.